t/ui/25-developer_mode.t

```diff
@@ -447,34 +463,60 @@ subtest 'start developer session' => sub {
     };
 
     subtest 'select whether to pause on assert_screen failure' => sub {
-        my $checkbox = $driver->find_element('#developer-pause-on-timeout');
-        is($checkbox->is_selected, 0, 'check box initially not selected');
+        my @options = $driver->find_elements('#developer-pause-on-mismatch option');
+        is(scalar @options,            3, 'three options for pausing on screen mismatch');
+        is($options[0]->is_selected(), 1, 'pausing on screen mismatch disabled by default');
+
+        # attempt to turn pausing on assert_screen on when current state unknown
+        $options[1]->set_selected();
+        assert_sent_commands(undef, 'prevent overriding pause on mismatch with form defaults if unknown');
+
+        # assume we know the pausing on screen match is disabled
+        fake_state(developerMode => {pauseOnScreenMismatch => 'false'});
+        is($options[0]->is_selected(), 1, 'pausing on screen mismatch still disabled');
 
         # turn pausing on assert_screen on
-        $checkbox->click();
+        $options[1]->set_selected();
         assert_sent_commands(
             [
                 {
-                    cmd  => 'set_pause_on_assert_screen_timeout',
-                    flag => 1,
+                    cmd      => 'set_pause_on_screen_mismatch',
+                    pause_on => 'assert_screen',
                 }
             ],
             'command to pause on assert_screen failure sent'
         );
 
         # fake the feedback from os-autoinst
-        fake_state(developerMode => {pauseAtTimeout => 1});
+        fake_state(developerMode => {pauseOnScreenMismatch => '"assert_screen"'});
+        is($options[1]->is_selected(), 1, 'still right option selected');
 
-        # turn pausing on assert_screen off
-        $checkbox->click();
+        # turn pausing on check_screen on
+        $options[2]->set_selected();
         assert_sent_commands(
             [
                 {
-                    cmd  => 'set_pause_on_assert_screen_timeout',
-                    flag => 0,
+                    cmd      => 'set_pause_on_screen_mismatch',
+                    pause_on => 'check_screen',
                 }
             ],
-            'command to unset pause on assert_screen failure sent'
+            'command to pause on check_screen failure sent'
+        );
+
+        # fake the feedback from os-autoinst
+        fake_state(developerMode => {pauseOnScreenMismatch => '"check_screen"'});
+        is($options[2]->is_selected(), 1, 'still right option selected');
+
+        # turn pausing on screen mismatch off
+        $options[0]->set_selected();
+        assert_sent_commands(
+            [
+                {
+                    cmd      => 'set_pause_on_screen_mismatch',
+                    pause_on => undef,
+                }
+            ],
+            'command to turn pausing on screen mismatch off sent'
```

Comment:
just regarding the formatting, is it tidy that spreads out over 8 lines? Maybe you can write like:

```suggestion
        my $cmds = {cmd => 'set_pause_on_screen_mismatch', pause_on => undef};
        assert_sent_commands([$cmds], 'command to turn pausing on screen mismatch off sent');
```

Reply:
I still think it is more readable without introducing an extra variable for each occurrence of that pattern.